Framework/Core/src/FrameworkGUIDebugger.cxx

```diff
@@ -100,6 +104,117 @@ ImVec4 colorForLogLevel(LogParsingHelpers::LogLevel logLevel)
   };
 }
 
+// This is to display the information in the data relayer
+struct HeatMapHelper {
+  template <typename RECORD, typename ITEM>
+  static void draw(const char* name,
+                   float widgetYSize,
+                   std::function<size_t()> const& getNumRecords,
+                   std::function<RECORD(size_t)> const& getRecord,
+                   std::function<size_t(RECORD const&)> const& getNumItems,
+                   std::function<ITEM const&(RECORD const&, size_t)> const& getItem,
+                   std::function<int(ITEM const&)> const& getValue,
+                   std::function<ImU32(int value)> const& getColor)
+  {
+    ImU32 BORDER_COLOR = ImColor(200, 200, 200, 40);
+    ImDrawList* drawList = ImGui::GetWindowDrawList();
+    ImVec2 winPos = ImGui::GetCursorScreenPos();
+    ImVec2 canvas_sz = ImGui::GetWindowSize();
+    ImGui::BeginChild(name, ImVec2(canvas_sz.x, widgetYSize), true);
+    drawList->AddQuad(
+      ImVec2(0., 0.) + winPos,
+      ImVec2{ canvas_sz.x - 1, 0 } + winPos,
+      ImVec2{ canvas_sz.x - 1, widgetYSize } + winPos,
+      ImVec2{ 0, widgetYSize } + winPos,
+      BORDER_COLOR);
+    float padding = 1;
+    for (size_t ri = 0, re = getNumRecords(); ri < re; ri++) {
+      auto record = getRecord(ri);
+      ImVec2 xOffset{ (ri * canvas_sz.x / getNumRecords()) + padding, 0 };
+      ImVec2 xSize{ canvas_sz.x / getNumRecords() - 2 * padding, 0 };
+      for (size_t mi = 0, me = getNumItems(record); mi < me; mi++) {
+        ImVec2 yOffSet{ 0, (mi * widgetYSize / getNumItems(record)) + padding };
+        ImVec2 ySize{ 0, (widgetYSize / getNumItems(record)) - 2 * padding };
+        drawList->AddQuadFilled(
+          xOffset + yOffSet + winPos,
+          xOffset + xSize + yOffSet + winPos,
+          xOffset + xSize + yOffSet + ySize + winPos,
+          xOffset + yOffSet + ySize + winPos,
+          getColor(getValue(getItem(record, mi))));
+      }
+    }
+    ImGui::EndChild();
+  }
+};
+
+bool startsWith(std::string mainStr, std::string toMatch)
+{
+  if (mainStr.find(toMatch) == 0) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+void displayDataRelayer(DeviceMetricsInfo const& metrics,
+                        DeviceInfo const& info)
```

Comment:
get a compiler warning here
```
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/FrameworkGUIDebugger.cxx:183:23: warning: declaration of ‘const o2::framework::MetricInfo& info’ shadows a parameter [-Wshadow]
     MetricInfo const& info = metrics.metrics[viewIndex.indexes[idx]];
                       ^~~~
/home/richter/src/alisw/sw/SOURCES/O2/dev/0/Framework/Core/src/FrameworkGUIDebugger.cxx:160:43: note: shadowed declaration is here
                         DeviceInfo const& info)
                                           ^~~~
```
I thought warnings are treated as errors in the CI, has this policy been policy changed? @dberzano , @sawenzel